lib/PuppeteerSharp/Page.cs

```diff
@@ -200,11 +200,17 @@ private Page(Session client, Target target, FrameTree frameTree, bool ignoreHTTP
             return await context.QueryObjects(prototypeHandle);
         }
 
-        public async Task<object> EvalAsync(string selector, Func<object> pageFunction, params object[] args)
-            => await MainFrame.Eval(selector, pageFunction, args);
-
-        public async Task<object> EvalAsync(string selector, string pageFunction, params object[] args)
-            => await MainFrame.Eval(selector, pageFunction, args);
+        /// <summary>
+        /// Runs <c>document.querySelector</c> within the frame and passes it as the first argument to <c>pageFunction</c>
+        /// </summary>
+        /// <typeparam name="T">The type of the response</typeparam>
+        /// <param name="selector">A <c>selector</c> to query frame for</param>
+        /// <param name="pageFunction">Function to be evaluated in browser context</param>
+        /// <param name="args">Arguments to pass to <c>pageFunction</c></param>
+        /// <exception cref="PuppeteerException">If there's no element matching <paramref name="selector"/></exception>
+        /// <returns>Task which resolves to the return value of <c>pageFunction</c></returns>
+        public Task<T> EvalAsync<T>(string selector, string pageFunction, params object[] args)
+            => MainFrame.EvalAsync<T>(selector, pageFunction, args);
```

Comment:
I'd like us to think about this name @Meir017.
We weren't able to use `$` so we used `GetElement`. It looks like something is missing in this name.
I'm not saying that this would be the right answer but, how about `EvalElementAsync`?

Reply:
well, it doesn't evaluate the element. it evaluates a functions where the first argument is the element